docs/interpretation_of_JSON_Schema_draft_7.md

```diff
@@ -34,21 +35,28 @@ Currently the following `not` model properties are interpreted:
 - boolean `not` schemas are not applied.
 
 ## allOf sub schemas
-`allOf` are a bit different then the other [combination keywords](#Processing-sub-schemas) since it can imply inheritance. 
+`allOf` is a bit different than the other [combination keywords](#Processing-sub-schemas) since it can imply inheritance. 
 
-So dependant on whether the simplify option `allowInheritance` is true or false we interpret it as inheritance or simply merge the models together.
-
-## Determining the type for the model
-To determine the types for the model we use the following interpretation (and in that order):
-- `true` schema infers all model types (`object`, `string`, `number`, `array`, `boolean`, `null`, `integer`).
-- Usage of `type` infers the initial model type.
-- Usage of `properties` infers `object` model type.
-- Usage of `const` infers the constant value as type if schema does not have `type` specified.
-- Usage of `enum` infers the enum values as type if schema does not have `type` specified.
+So dependant on whether the interpreter option `allowInheritance` is true or false we interpret it as inheritance or [merge](#Merging-models) the models.
 
 ## Processing sub schemas
-The following JSON Schema keywords are merged with the already interpreted model:
+The following JSON Schema keywords are [merged](#Merging-models) with the already interpreted model:
 - `oneOf`
 - `anyOf`
 - `then`
 - `else`
+
+
+## Merging models
+Because of the recursive nature of the interpreter (and the nested behavior of JSON Schema) it happens that two models needs to be merged together.
+
+If only one side has a property defined, it is used as is, if both have it defined they are merged based on the following logic:
+- `additionalProperties` if both models contain it the two are recursively merged together. 
+- `patternProperties` if both models contain a pattern the corresponding models are recursively merged together. 
+- `properties` if both models contain the same property the corresponding models are recursively merged together. 
+- `items` if both models contain items they are recursively merged together.
+- `types` if both models contain types they are merged together, duplicate types are removed.
+- `enum` if both models contain enums they are merged together, duplicate enums are removed.
+- `required` if both models contain required properties they are merged together, duplicate required properties are removed.
+- `$id`, `$ref`, `extend` uses left model value if present otherwise right model value if present.
+- `originalSchema` are overwritten.
```

Comment:
We have information about `originalSchema` only in `API.md` file, we should described what is `originalSchema` here in another place :)

Reply:
Hmm, yea true. What do you think is the best solution? 🤔

Reply:
It can be described here, or somewhere at the beginning of this doc :)

Reply:
Added a link to the input processor docs which explains the CommonModel, otherwise, we have to keep in mind that we have a description of a property in this document as well.